trieste/utils/multi_objectives.py

```diff
@@ -0,0 +1,232 @@
+    See :cite:`van1999multiobjective` and  :cite: fonseca1995multiobjective
+    (the latter for discussion of pareto front property) for details.
+    """
+
+    bounds = [[-2.0] * 2, [2.0] * 2]
+    dim = 2
+
+    def objective(self):
+        return vlmop2
+
+    def gen_pareto_optimal_points(self, n: int, seed=None):
+        tf.debugging.assert_greater(n, 0)
+        _x = tf.linspace([-1 / tf.sqrt(2.0)], [1 / tf.sqrt(2.0)], n)
+        return vlmop2(tf.concat([_x, _x], axis=1))
+
+
+def vlmop2(x: TensorType) -> TensorType:
+    """
+    The VLMOP2 synthetic function.
+
+    :param x: The points at which to evaluate the function, with shape [..., 2].
+    :return: The function values at ``x``, with shape [..., 2].
+    :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
+    """
+    tf.debugging.assert_shapes([(x, (..., 2))], message="vlmop2 only allow 2d input")
+    transl = 1 / math.sqrt(2.0)
+    y1 = 1 - tf.exp(-1 * tf.reduce_sum((x - transl) ** 2, axis=-1))
+    y2 = 1 - tf.exp(-1 * tf.reduce_sum((x + transl) ** 2, axis=-1))
+    return tf.stack([y1, y2], axis=-1)
+
+
+class DTLZ(MultiObjectiveTestProblem):
+    """
+    DTLZ series multi-objective test problem.
+    See :cite:deb2002scalable for details.
+    """
+
+    def __init__(self, input_dim: int, num_objective: int):
+        tf.debugging.assert_greater(input_dim, 0)
+        tf.debugging.assert_greater(num_objective, 0)
+        tf.debugging.assert_greater(
+            input_dim,
+            num_objective,
+            f"input dimension {input_dim}"
+            f"  must be greater than function objective numbers {num_objective}",
+        )
+        self._dim = input_dim
+        self.M = num_objective
+        self.k = self._dim - self.M + 1
+        self._bounds = [[0] * input_dim, [1] * input_dim]
+
+    @property
+    def dim(self):
+        return self._dim
+
+    @property
+    def bounds(self):
+        return self._bounds
+
+
+class DTLZ1(DTLZ):
+    """
+    The DTLZ1 problem, the idea pareto fronts lie on a linear hyper-plane.
+    See :cite:deb2002scalable for details.
+    """
+
+    def objective(self):
+        return partial(dtlz1, m=self.M, k=self.k, d=self.dim)
+
+    def gen_pareto_optimal_points(self, n: int, seed=None):
+        tf.debugging.assert_greater_equal(self.M, 2)
+        rnd = tf.random.uniform([n, self.M - 1], minval=0, maxval=1, seed=seed)
+        strnd = tf.sort(rnd, axis=-1)
+        strnd = tf.concat([tf.zeros([n, 1]), strnd, tf.ones([n, 1])], axis=-1)
+        return 0.5 * (strnd[..., 1:] - strnd[..., :-1])
+
+
+def dtlz1(x: TensorType, m: int, k: int, d: int) -> TensorType:
+    """
+    The DTLZ1 synthetic function.
+
+    :param x: The points at which to evaluate the function, with shape [..., d].
+    :param m: The objective numbers.
+    :param k: The input dimensionality for g.
+    :param d: The dimensionality of the synthetic function.
+    :return : The function values at ``x``, with shape [..., m].
```

Comment:
(I think the docs check is failing because of this, and the similar one further down)

```suggestion
    :return: The function values at ``x``, with shape [..., m].
```

Reply:
Thanks, this seems not very clear to learn from error message like:
`/home/runner/work/trieste/trieste/docs/autoapi/trieste/utils/multi_objectives/index.rst:147:Field list ends without a blank line; unexpected unindent.`